sql/catalyst/src/main/java/org/apache/spark/sql/catalog/v2/RequiresSessionCatalog.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalog.v2;
+
+import org.apache.spark.annotation.Unstable;
+import org.apache.spark.sql.catalyst.catalog.SessionCatalog;
+
+/**
+ * A marker interface to take the Spark internal session catalog, so that the catalog implementation
+ * can hook in some custom logic before Spark calling the session catalog.
+ */
+@Unstable
+public interface RequiresSessionCatalog {
+  void initWithSessionCatalog(SessionCatalog catalog);
```

Comment:
If this doesn't replace the existing catalog initialization method, then it should have a different name. How about `setSessionCatalog`?